python/shotgun_model/shotgun_hierarchy_model.py

```diff
@@ -498,6 +543,9 @@ def _refresh_data(self):
         view related states such as selection.
         """
 
+        if not self._hierarchy_is_supported:
+            return
```

Comment:
Should be `return None` since you're returning actual data at the end.

https://www.python.org/dev/peps/pep-0008/#programming-recommendations